File: api/Services/ArticleService.cs
namespace WebApi.Services;

using AutoMapper;
using BCrypt.Net;
using WebApi.Entities;
using WebApi.Helpers;
using WebApi.Models.Article;
using WebApi.Repositories;

public interface IArticleService
{
    Task<IEnumerable<Article>> GetAll();
    Task<Article> GetById(int id);
    Task Create(CreateRequest model);
    Task Update(int id, UpdateRequest model);
    Task Delete(int id);
}

public class ArticleService : IArticleService
{
    private IArticleRepository _articleRepository;
    private readonly IMapper _mapper;

    public ArticleService(
        IArticleRepository articleRepository,
        IMapper mapper)
    {
        _articleRepository = articleRepository;
        _mapper = mapper;
    }

    public async Task<IEnumerable<Article>> GetAll()
    {
        return await _articleRepository.GetAll();
    }

    public async Task<Article> GetById(int id)
    {
        var article = await _articleRepository.GetById(id);

        if (article == null)
            throw new KeyNotFoundException("Article not found");

        return article;
    }

    public async Task Create(CreateRequest model)
    {
        var article = _mapper.Map<Article>(model);
        await _articleRepository.Create(article);
    }

    public async Task Update(int id, UpdateRequest model)
    {
        var article = await _articleRepository.GetById(id);

        if (article == null)
            throw new KeyNotFoundException("Article not found");

        _mapper.Map(model, article);

        await _articleRepository.Update(article);
    }

    public async Task Delete(int id)
    {
        await _articleRepository.Delete(id);
    }
}

File: front/src/_services/article.service.js
import config from 'config';
import { fetchWrapper } from '@/_helpers';

const baseUrl = `${config.apiUrl}/articles`;

export const articleService = {
    getAll,
    getById,
    create,
    update,
    delete: _delete
};

function getAll() {
    return fetchWrapper.get(baseUrl);
}

function getById(id) {
    return fetchWrapper.get(`${baseUrl}/${id}`);
}

function create(params) {
    return fetchWrapper.post(baseUrl, params);
}

function update(id, params) {
    return fetchWrapper.put(`${baseUrl}/${id}`, params);
}

function _delete(id) {
    return fetchWrapper.delete(`${baseUrl}/${id}`);
}
